kobo-uncaged/device/device.go

```diff
@@ -481,6 +479,82 @@ func (k *Kobo) SaveCoverImage(contentID string, size image.Point, imgB64 string)
 	}
 }
 
+// AddMetaByTrigger allows KU to use a DB trigger to update book metadata
+// when Nickel imports book content
+func (k *Kobo) AddMetaByTrigger() error {
+	var err error
+	// Table to hold temporary metadata for the trigger to use
+	metaTableQuery := `
+		CREATE TABLE IF NOT EXISTS _ku_meta (
+			ContentID    TEXT NOT NULL UNIQUE,
+			Description  TEXT,
+			Series       TEXT,
+			SeriesNumber TEXT,
+			PRIMARY KEY(ContentID)
+		);`
+	if _, err = k.nickelDB.Exec(metaTableQuery); err != nil {
+		return fmt.Errorf("AddMetaByTrigger: Create table error: %w", err)
+	}
+	// Trigger fired when Nickel inserts a book into the content table
+	// It replaces and/or adds metadata after the record has been inserted
+	triggerQuery := `
+		DROP TRIGGER IF EXISTS _ku_meta_content_insert;
+		CREATE TRIGGER _ku_meta_content_insert
+			AFTER INSERT ON content WHEN
```

Comment:
Note that this will only work on books which haven't been imported yet.

Reply:
Also, it won't appear in nickel until it's restarted (and if you open the book, you'll lose the changes, unless you have an update hook too).

Reply:
Ahh, you're right, forgot about the update existing metadata case.

Reply:
And if you have an update trigger, it shouldn't conflict with seriesmeta since triggers don't trigger triggers by default.

Reply:
BTW, KU automatically triggers another USBMS session after content import to ensure that the metadata is updated by Nickel

Reply:
The other option would be an INSTEAD OF hook, but you'd need to ensure you cover all columns in the Content table to prevent breaking nickel (and it'd be version-dependent, which I wanted to avoid in seriesmeta).

Reply:
I'll probably directly update metadata for content that already exists. No need for a trigger to do that. I had just forgot about it. Thanks for the reminder!

Reply:
Yes. I did it all with triggers in kepubify since I was using them anyways (and had an update trigger), but this doesn't need that.